leenlab2/reminiscence
# Teleportation and move puzzle targets

Move puzzle targets to memory palace and allow player to go back and forth between memory palace and main real life attic.

# Description

- Moved puzzle targets into new memory palace
- Added button on TV to enter memory if a tape is in the TV
- Added Memory Scene Canvas to display "Tape rewriting in progress..." if the player places a branching item in the right place, or all 3 key items in the right place
- Teleport player out upon branching item placed in right spot

A few future improvements:
- teleport player back to attic relative to TV(right now using absolute positioning)
- Add vhs background when tape rewrites
- Play error noise if wrong item is placed in wrong shadow cue

### Type of change






New Feature
## How to Integrate

This is already integrated in main scene
# How Has This Been Tested?

- Open the TV with no tape inserted and press the button to enter the memory. Nothing should happen
- Insert a tape and enter the memory. Confirm the effects of the camera are there. Confirm you cannot open the TV while inside the memory. Exit the memory by pressing TAB on keyboard on ButtonEast on controller
- Pick up one poster and enter the scene again. Place the poster in the right location. You should see "Tape rewriting in progress" and be transported out after 4 seconds back to the TV. Ensure the correct tape is playing.
- Go back into the memory and confirm the 3 shadow cues are present.
- Transport the 3 items to the correct spots from the attic into the memory. Confirmation noise will play once all 3 items are placed. You can exit back to the attic to watch the completed tape.
- 
# Checklists
## Integration Checklist
- x ] I have attached this PR to the relevant Trello card

## Review Checklist
